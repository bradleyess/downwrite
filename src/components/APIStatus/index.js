import React from 'react'
import { Block } from 'glamor/jsxstyle'

const pos = {
  bottom: 8,
  right: 8,
  position: 'fixed'
}

export default ({ data, env }) => (
  <Block
    fontSize={12}
    opacity='.375'
    background='#CF4832'
    color='white'
    textAlign='center'
    padding={8}
    {...pos}>
    {data && (
      <span>
        {data.post} {env}
      </span>
    )}
  </Block>
)
