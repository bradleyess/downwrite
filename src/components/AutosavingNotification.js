import React from 'react'
import { Block } from 'glamor/jsxstyle'

/* TypeError: cyclic object value */
export default () => (
	<Block
		width={160}
		textAlign="center"
		top={20}
		right={0}
		left={0}
		zIndex="900"
		position="fixed"
		margin="auto"
		backgroundColor="white"
		boxShadow="0 0 2px rgba(0,0,0,.07), 0 2px 4px rgba(0,0,0,.12)"
		padding={4}>
		Autosaving
	</Block>
)
