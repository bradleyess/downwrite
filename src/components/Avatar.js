import React from 'react'
import { Block } from 'glamor/jsxstyle'

class RandomColor extends React.Component {
	state = {
		a: null,
		b: null
	}

	componentWillMount() {
		this.setState({
			a: `#FEB692`,
			b: `#EA5455`
		})
	}

	shouldComponentUpdate() {}

	render() {
		return this.props.children(this.state.a, this.state.b)
	}
}

export default () => (
	<RandomColor>
		{(A, B) => (
			<Block
				borderRadius={32}
				width={32}
				height={32}
				background={`linear-gradient(135deg, ${A} 10%, ${B} 100%)`}
				margin="auto"
			/>
		)}
	</RandomColor>
)
