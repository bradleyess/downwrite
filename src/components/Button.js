// @flow

import React from 'react'
import { css } from 'glamor'

const btn = css({
	display: 'block',
	fontSize: 14,
	backgroundColor: 'var(--color-6)',
	transition: 'background-color 250ms ease-in-out',
	color: '#282828',
	fontWeight: 700,
	border: 0,
	paddingTop: 4,
	paddingLeft: 18,
	paddingRight: 18,
	paddingBottom: 4,
	borderRadius: 4,
	boxShadow: '0 0 4px rgba(0,0,0,.037), 0 4px 8px rgba(0,0,0,.07)',
	':hover': {
		backgroundColor: 'var(--color-7)'
	}
})

type Props = {
	args: Object
}

const Button = ({ ...args }: Props) => <button className={css(btn)} {...args} />

export default Button
