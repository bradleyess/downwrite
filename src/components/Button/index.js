import React from 'react'
import { css } from 'glamor'

const BUTTON_SIZE = 62.5

const btn = css({
  display: 'block',
  backgroundColor: '#2584A5',
  transition: 'background-color 250ms ease-in-out',
  color: 'white',
  border: 0,
  padding: 16,
  borderRadius: '50%',
  maxHeight: BUTTON_SIZE,
  width: BUTTON_SIZE,
  height: '100%',
  boxShadow: '0 0 4px rgba(0,0,0,.07), 0 4px 8px rgba(0,0,0,.14)',
  ':hover': {
    backgroundColor: 'var(--color-2)'
  }
})

const positionedSty = css({
  position: 'absolute',
  left: -32,
  top: -32
})

const Button = ({ onClick, positioned, type, ...args }) => {
  let btnStyle = css(
    positioned ? [btn, positionedSty] : btn
  )

  return (
    <button onClick={onClick} className={btnStyle} {...args} />
  )
}

export default Button
