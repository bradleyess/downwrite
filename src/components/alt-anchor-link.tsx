import * as React from "react";
import classNames from "../utils/classnames";

interface IAltAnchorProps<T> extends React.AnchorHTMLAttributes<T> {
  space?: string;
}

export default React.forwardRef<HTMLAnchorElement, IAltAnchorProps<unknown>>(
  (props: IAltAnchorProps<unknown>, ref) => {
    const cx = classNames("AltLink", props.className);
    return (
      <a href={props.href} className={cx} style={props.style} ref={ref}>
        {props.children}
      </a>
    );
  }
);
