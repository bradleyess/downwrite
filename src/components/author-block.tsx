import * as React from "react";
import { Link } from "react-router-dom";
import Avatar from "./avatar";

interface IAuthorProps {
  authed: boolean;
  name: string;
  colors: string[];
}

export default function Author(props: IAuthorProps): JSX.Element {
  return (
    <aside className="Sheet AuthorBlock Wrapper Wrapper--sm">
      <header className="AuthorBlockHeader">
        <Avatar colors={props.colors} />
        <h6 className="AuthorBlockTitle">Post from {props.name}</h6>
      </header>
      {!props.authed && (
        <React.Fragment>
          <hr />
          <p className="AuthorBlockContent">
            <span>
              You can write and share on Downwrite, you can sign up or log in{" "}
            </span>
            <Link to="/login">here</Link>
          </p>
        </React.Fragment>
      )}
    </aside>
  );
}
