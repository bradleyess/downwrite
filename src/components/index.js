import Button from './Button'
import Card from './Card'
import DWEditor from './Draft'
import EmptyPosts from './EmptyPosts'
import Header from './Header'
import Input from './Input'
import Loading from './Loading'
import Logo from './Logo'
import Nav from './Nav'
import PostList from './PostList'
import Toolbar from './Toolbar'
import Wrapper from './Wrapper'
import Helpers from './Helpers'

export {
	Helpers,
	DWEditor,
	Wrapper,
	Toolbar,
	Button,
	Card,
	EmptyPosts,
	Header,
	Input,
	Loading,
	Logo,
	Nav,
	PostList
}
