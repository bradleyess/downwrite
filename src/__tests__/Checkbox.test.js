import { Check } from '../components'

const checkFire = jest.fn()

let wrapper = mount(<label><Check checked onChange={checkFire} /></label>)

describe('<Check />', () => {
	xit('checks', () => {
		wrapper.find('input').simulate('toggle')
		expect(checkFire).toHaveBeenCalled()
	}) 

	it('matches snapshot', () => expect(wrapper).toMatchSnapshot())
})
