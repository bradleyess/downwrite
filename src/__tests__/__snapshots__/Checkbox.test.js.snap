// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Check /> matches snapshot 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  border-radius: 4px;
  vertical-align: middle;
  width: 20px;
  height: 20px;
  appearance: none;
  border: 0;
  background: var(--color-1) url(check.svg) no-repeat center center;
  -webkit-appearance: none;
  -moz-appearance: none;
}

<label>
  <Component
    checked={true}
    onChange={[Function]}
  >
    <input
      checked={true}
      className={
        Object {
          "data-glamor-0": "",
        }
      }
      onChange={[Function]}
      type="checkbox"
    />
  </Component>
</label>
`;
